pkg/tenant/tenants.go

```diff
@@ -405,30 +404,7 @@ func (s *Server) TenantStats(c *gin.Context) {
 			return
 		}
 		totalTopics += len(topics)
-
-		// API keys are stored in Quarterdeck
-		req := &qd.APIPageQuery{
-			ProjectID: project.ID.String(),
-			PageSize:  100,
-		}
-
-		// We will always retrieve at least one page; it's possible but unlikely for a
-		// project to have more than 100 API keys.
-	keysLoop:
-		for {
-			var page *qd.APIKeyList
-			if page, err = s.quarterdeck.APIKeyList(ctx, req); err != nil {
-				sentry.Debug(c).Err(err).Msg("tracing quarterdeck error in tenant")
-				api.ReplyQuarterdeckError(c, err)
-				return
-			}
-			totalKeys += len(page.APIKeys)
-
-			if page.NextPageToken == "" {
-				break keysLoop
-			}
-			req.NextPageToken = page.NextPageToken
-		}
+		totalKeys += int(project.APIKeys)
```

Comment:
Here is the relevant change, `project.APIKeys` is updated asynchrously whenever API keys are created or deleted via `UpdateProjectStats` so no need to call Quarterdeck anymore here.